Homework4/Tests/SimpleFTPTests.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 Ivan Zakarlyuka.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace SimpleFTPTests;
+
+using System.Net;
+using SimpleFTP;
+
+public class Tests
+{
+
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+    private Server server;
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+    private Client client;
+#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
+    private IPEndPoint endPoint = new (IPAddress.Loopback, 8888);
+    [OneTimeSetUp]
+    public void OneTimeSetUp()
+    {
+        this.server = new (endPoint.Port);
+        this.client = new (endPoint);
+        _ = Task.Run(() => this.server.Start());
+    }
+
+    [OneTimeTearDown]
+    public void OneTimeTearDown() {
+        this.server.Stop();
+    }
+
+    [Test]
+    public async Task ListCorrectPathTest()
+    {
+        var a = await this.client.List("../../../TestFiles/test1");
+        var excpected = "2 ../../../TestFiles/test1/test1.txt false ../../../TestFiles/test1/test1folder true";

```

Comment:
```suggestion
        var expected = "2 ../../../TestFiles/test1/test1.txt false ../../../TestFiles/test1/test1folder true";

```
Кстати, можно не писать такие страшные пути к файлам, если выставлять опцию "copy to output directory"